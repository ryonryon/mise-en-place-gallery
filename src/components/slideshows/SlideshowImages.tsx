import styled from "@emotion/styled";
import * as React from "react";

interface Props {
  imgSrcs: string[];
  activeNumber: number;
}

const ImgGroupDiv = styled.div`
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  text-align: center;

  img {
    width: 100%;
    height: 100%;
  }

  .img-inactive {
    display: none;
  }
`;

const SlideshowImages = ({ imgSrcs, activeNumber }: Props) => (
  <ImgGroupDiv>
    {imgSrcs.map((imgSrc, index) => (
      <img
        key={index}
        src={imgSrc}
        alt={`img-${index}`}
        className={`img ${index === activeNumber ? "" : "img-inactive"}`}
      />
    ))}
  </ImgGroupDiv>
);

SlideshowImages.defaultProps = {
  activeNumber: 0
};

export default SlideshowImages;
