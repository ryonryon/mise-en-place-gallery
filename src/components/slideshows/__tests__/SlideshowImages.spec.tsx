import * as React from "react";
import * as renderer from "react-test-renderer";

import SlideshowImages from "../SlideshowImages";

describe("Arrow test", () => {
  it("tests Arrow render correctly", () => {
    const slideshowImages = renderer.create(<SlideshowImages imgSrcs={["test", "test"]} />);

    expect(slideshowImages).toMatchSnapshot();
  });
});
