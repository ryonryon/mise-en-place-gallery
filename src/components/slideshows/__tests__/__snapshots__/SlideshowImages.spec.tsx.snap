// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow test tests Arrow render correctly 1`] = `
<div
  className="css-tpx3lo"
>
  <img
    alt="img-0"
    className="img "
    src="test"
  />
  <img
    alt="img-1"
    className="img img-inactive"
    src="test"
  />
</div>
`;
